src/lib/hooks/useUpgradeDialog.js

```diff
@@ -35,8 +36,15 @@ export default () => {
   const [showDialog] = useDialog()
   const store = SimpleStore.useStore()
   const serviceWorkerUpdated = store.get('serviceWorkerUpdated')
+  const actualPhaseRef = useRef(null)
```

Comment:
maybe initialize to Promse.resolve(Config.phase)

Reply:
used observables instead. 
ref is initialized with observable having full API calling flow described. 
observables aren't executes until first subscription comes

so we have just declaration of how we would get data
then, on mount we're subscribing which initiales the flow declared and performs actual API call
if serviceWorkerUpdated changes before on mount hook executed, it also will perform API call because also uses .subscribe
the share() operator guarantees that our flow including API call will be executed once